### PR TITLE
Add support for POSTing to /series endpoint

### LINKED ISSUE #5417: [Feature Request] Support POST method for series endpoint
## Proposal
**Use case. Why is this important?**
Some user queries `/api/v1/series` with many label values, it will trigger the GET url length limit.

If support POST method for `/api/v1/series` like `/api/v1/query` endpoint, the use case will be fine.

### PR DESCRIPTION
**Pull request by bobmshannon**
Closes #5417

The [ParseForm()](https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L446) method in [net/http](https://golang.org/pkg/net/http/#Request.ParseForm) supports url-encoded parameters in the request body if the method is `POST`, with a built-in 10 MB limit.

This change should mitigate issues where a user runs into query string length limits when providing a large number of selectors using the `GET` method inline in the URL query string.

### CODE REVIEW

**Comment by eahydra** [comment]
maybe need modify web/api/v1/api_test.go L838, support POST unit test for series endpoint.

**Comment by eahydra** [comment]
LGTM

File: docs/querying/api.md [hunk 1]
```diff
@@ -201,10 +201,11 @@ $ curl 'http://localhost:9090/api/v1/query_range?query=up&start=2015-07-01T20:10
 
 ### Finding series by label matchers
 
-The following endpoint returns the list of time series that match a certain label set.
+The following endpoints return the list of time series that match a certain label set.
```
Comment (brian-brazil) [inline docs/querying/api.md]: It's still one endpoint, but multiple verbs

**Comment by bobmshannon** [comment]
I was going to suggest adding POST support to the query endpoints for https://github.com/prometheus/client_golang/issues/428, but it looks like that was already implemented. Instead, I've just updated the API documentation instead.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**